platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ExportMigrationContextImpl.java

```diff
@@ -244,6 +243,27 @@ private boolean isDirectory(ExportMigrationEntryImpl entry) {
     }
     return true;
   }
+
```

Comment:
❓ This is really better than simply suppressing that warning?

Reply:
I don't love that it makes the formatting ugly, but I'd really like us to avoid developing a habit of suppressing when we don't have to.

Reply:
My issue here is that it adds code that shouldn't be there to start with. This ends up lowering the coverage and it adds maintenance. I would rather keep the suppression. As for not suppressing when we don't need to, I am in agreement. Maybe there are cases we can revisit the set of rules we have.

Reply:
The other part I don't like about doing this here is that the next developer will come in and thing we forgot to check these additional parameters. The suppression comment added clarification to why it was done that way.

Reply:
1. Sorry. When I replied about formatting, I mistakenly thought your initial comment was on the empty consumer.
2. This does lower coverage by adding lines of code, but that's an arbitrary decrease to be honest. It isn't _semantically_ less covered than it was before; Jacoco just isn't smart enough to know that.
3. I disagree that it adds maintenance. It adds methods that _should_ not need be touched again in order to make explicit the fact that the equality of this class is _not_ dependent on any of the fields present in it. The class-wide suppression could be missed by a developer in the future who might think this class _should_ include its own fields in the calculation of equality/hashcode (as would be the normal case) and might add them, thereby breaking logic. The suppression would then still be present, incorrectly, while the functionality was in fact wrong. Adding these method explicitly makes intent clear.